### ISSUE TITLE
songrec recognize hangs if it can't find a match on Shazam on macOS Big Sur

### ISSUE THREAD

**Issue by msdos** [marin-m/SongRec#93]
A random recording of complete gibberish, just for testing:

`songrec recognize ~/Downloads/random_25204.mp3`

This hangs the terminal. The command never finishes.

Different command (I've stripped the real values, with an exception to `matches` that comes as is):

```
songrec audio-file-to-recognized-song ~/Downloads/random_25204.mp3

{
  "location": {
    "accuracy": 0.00,
    "altitude": 000.0,
    "latitude": 00.0,
    "longitude": 0.0
  },
  "matches": [],
  "tagid": "UID",
  "timestamp": XXXXXXXX,
  "timezone": "XXXXXXXXX"
}

```

Don't know if this happens in other platforms. I'm not familiar with rust but it seems that `matches` should be verified for `[]` and finish the command as well (just don't know if should return a value different from 0 to shell. IMHO, songrec is expected to return something from Shazam, if nothing was found 1 should be returned, like grep does).

**Comment by awsms** [comment]
Same on Linux

**Comment by marin-m** [comment]
Thanks, should be addressed in commit 9bef24e273098a1380be8a46cbaf2692daad8a5b

**Comment by msdos** [comment]
Just downloaded a random wav from https://www.random.org/audio-noise/ and converted to mp3:

```
songrec recognize soundfile.mp3                                                                                                                                                      
Error: No match for this song
```

@awsms can you update and check on your Linux as well? So we can close this issue.
